score-xml/src/main/java/io/cloudslang/content/xml/entities/inputs/CommonInputs.java

```diff
@@ -49,5 +143,76 @@ public CommonInputsBuilder withSecureProcessing(String inputValue) {
             secureProcessing = Boolean.parseBoolean(inputValue);
             return this;
         }
+
+        public CommonInputsBuilder withXmlDocumentSource(String xmlDocumentSource) {
+            if (xmlDocumentSource.isEmpty()) {
+                this.xmlDocumentSource = Constants.Defaults.XML_DOCUMENT_SOURCE;
+            } else if (Constants.Defaults.XML_DOCUMENT_SOURCE.equalsIgnoreCase(xmlDocumentSource) || Constants.XML_PATH.equalsIgnoreCase(xmlDocumentSource) || Constants.XML_URL.equalsIgnoreCase(xmlDocumentSource)){
+                this.xmlDocumentSource = xmlDocumentSource;
+            } else {
+                throw new RuntimeException(xmlDocumentSource + Constants.INVALID_XML_DOCUMENT_SOURCE);
+            }
+            return this;
+        }
+
+        public CommonInputsBuilder withUsername(String username) {
```

Comment:
@tethryus Usually in the builder you should do a basic validation of the input ... 
StringUtils.isBlank(some_none_required_input) ? defaultValue :  some_none_required_input ... 
WDYT?